yt/utilities/lib/ewah_bool_wrap.pyx

```diff
@@ -1747,32 +1740,20 @@ cdef class SparseUnorderedBitmaskSet:
     cdef void _fill(self, np.uint8_t[:] mask):
         cdef np.uint64_t ind
         cdef cset[np.uint64_t] *entries = <cset[np.uint64_t]*> self.entries
-        cdef cset[np.uint64_t].iterator it
-        it = entries[0].begin()
-        while it != entries[0].end():
-            ind = dereference(it)
-            mask[ind] = 1
-            preincrement(it)
+        for it in entries[0]:
```

Comment:
Here, and in a few other places, the iterator (it) is not declared. It also looks like there could be some clean up of unused variables (e.g. ind) that were only used when dereferencing pointers.

Reply:
This is true, it's not declared!  I checked into it, and it looks like in this particular case, it infers the type.  The generated code here looks like:

```
  __pyx_t_2 = &(__pyx_v_entries[0]);
  __pyx_t_1 = __pyx_t_2->begin();
  for (;;) {
    if (!(__pyx_t_1 != __pyx_t_2->end())) break;
    __pyx_t_3 = *__pyx_t_1;
    ++__pyx_t_1;
    __pyx_v_it = __pyx_t_3;
```

which is, I think, correct.  I believe it *has* to infer the type, because if it can't, it would not be able to cast the iter to a python object and would bail.

I'm working on the cleanup of unused vars!

Reply:
Looks good!